docs: signals and actions documentation (#6641)
Unleash/unleash: website/docs/reference/signals.mdx

---
title: Signals
---

:::info Availability

Unleash signals is an **enterprise-only** feature released as beta in Unleash v5.11.

:::

## Overview

> Signals enable external systems to communicate events to Unleash via a simple HTTP API.

Reviewer: ```suggestion
Signals represent that something happened somewhere. Signal endpoints allow external systems to let Unleash know about these signals via a simple HTTP API.
```

> Unleash can then configure automated actions to react to those events.

Reviewer: ```suggestion
You can then configure [actions](./actions) to react to the received signals automatically.
```

![Signals from external systems.](/img/signals/signals-input.png)

> You can create different endpoints to be used by different systems, or you can create a single endpoint and use different API tokens for different systems.

Reviewer: ```suggestion
Depending on your use case, you can create multiple signal endpoints, one per each external source, or re-use the same endpoint across different sources, using different API tokens for each, so the source can be identified.
```

## Creating signal endpoints

:::info Permissions

> Creating signal endpoints require admin permissions.

Reviewer: ```suggestion
Creating signal endpoints requires the `ADMIN` permission.
```

:::

To create a new signal endpoint, navigate to the **Admin UI** and select **Integrations** from the configure menu.

![Integrations menu.](/img/signals/integrations.png)

> In the integrations page, find the Signals integration:

Reviewer: ```suggestion
In the integrations page, find the Signals integration:

```

![Integrations menu.](/img/signals/signals-option.png)

> Clicking on the "New Signal" button will open a form where you can configure the signal endpoint.

Reviewer: ```suggestion
After accessing the Signals integration configuration, using the "New signal endpoint" button will open a form where you can configure your new signal endpoint.
```

![New signal form.](/img/signals/new-signal-form.png)

## Using a signal endpoint

> Using a signal endpoitn is as easy as sending a POST request to the signal endpoint URL with the correct API token. The body payload should be formatted as JSON.

Reviewer: ```suggestion
Using a signal endpoint is as easy as sending a POST request to the signal endpoint URL with the correct API token. The body payload is optional, but if present should be formatted as JSON.
```

```shell

> curl -X POST https://unleash.example.com/api/signals/my-signal-endpoint \

Reviewer: ```suggestion
curl -X POST https://unleash.example.com/api/signal-endpoint/my-signal-endpoint \
```

  -H "Authorization: Bearer your_token_secret" \
  -H "Content-Type: application/json" \
  -d '{

>     "type": "my-event",

Reviewer: ```suggestion
    "type": "my-signal",
```

    "data": {
      "key": "value"
    }
  }'
```

> Configuring an external system to use this API should be straightforward as there's no requirement on the payload structure.

Reviewer: ```suggestion
Configuring an external system to use your new signal endpoint should be straightforward as there's no requirement on the payload structure.
```

## View signals as they arrive

> To view the signals as they arrive, navigate to the **Signals** page in the **Admin UI** and select **View signals** option. 

Reviewer: ```suggestion
To view the signals as they arrive, navigate to the **Signals** page in the **Admin UI** and select the **View signals** option in the respective signal endpoint row.
```

![View signals option.](/img/signals/view-signals-option.png)

> This will help you test that the integration is working before you send in the real signals.

Reviewer: ```suggestion
This can help you test that the integration is working before you send in the real signals, or diagnose any unexpected behavior.
```

## Limits that apply

> There are some constraints on the signal endpoints as a meassure to protect our infrastructure. These values can be overriden for self-hosted installations.

Reviewer: ```suggestion
There are some constraints on the signal endpoints API as a measure to protect the stability of Unleash. These values can be overridden for self-hosted installations.
```

> - **Rate limit**: 10 requests per second.

Reviewer: ```suggestion
- **Rate limit**: 1 request per second.
```
